prow/cmd/deck/main.go

```diff
@@ -476,11 +477,18 @@ func handleLog(lc logClient) http.HandlerFunc {
 		w.Header().Set("Access-Control-Allow-Origin", "*")
 		job := r.URL.Query().Get("job")
 		id := r.URL.Query().Get("id")
-		if err := validateLogRequest(r); err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
+		podName := r.URL.Query().Get("podname")
```

Comment:
Could either make a different endpoint for this, or at the last verify that a user is providing a job + ID _OR_ the pod name

Reply:
As per your previous comment, would it be better to remove the option to specify a pod name in the endpoint altogether?

Reply:
I'm confused about the point at which a consumer knows the `Pod` name but doesn't know the job it was triggered for, but I trust you have such a use case. As it would be user error to provide the `Pod` name _and_ the job + ID, we should just validate that the user has made no mistakes. Either by checking query parameters here or by serving two endpoints. If you have no such case -- at all times when you know the `Pod` you know the job + ID, then just use the current endpoint.